validation/markets-validator.js

```diff
@@ -0,0 +1,549 @@
+			var feature = features[i];
+			this.validateFeature(feature);
+		}
+	};
+
+	this.validateFeature = function(feature) {
+		var featureValidator = new FeatureValidator(feature, this.cityName);
+		featureValidator.validate();
+		if (featureValidator.hasErrors()) {
+			this.errorsCount += featureValidator.getErrorsCount();
+		}
+		if (featureValidator.hasWarnings()) {
+			this.warningsCount += featureValidator.getWarningsCount();
+		}
+		this.featureValidators.push(featureValidator);
+	};
+
+	this.getErrorsCount = function() {
+		return this.errorsCount;
+	};
+
+	this.getWarningsCount = function() {
+		return this.warningsCount;
+	};
+
+	this.printIssues = function() {
+		for (var i = 0, length = this.featureValidators.length; i < length; ++i) {
+			var featureValidator = this.featureValidators[i];
+			featureValidator.printWarnings();
+			featureValidator.printErrors();
+		}
+	};
+
+}
+
+
+function FeatureValidator(feature, cityName) {
+
+	this.feature = feature;
+	this.cityName = cityName;
+	this.errors = [];
+	this.warnings = [];
+
+	this.validate = function() {
+		var feature = this.feature;
+		if (feature === undefined) {
+			this.errors.push(new CustomIssue("Feature cannot be undefined."));
+		} else if (feature === null) {
+			this.errors.push(new CustomIssue("Feature cannot be null."));
+		} else if (feature === {}) {
+			this.errors.push(new CustomIssue("Feature cannot be an empty object."));
+		} else {
+			this.validateGeometry(feature.geometry);
+			this.validateProperties(feature.properties);
+			this.validateType(feature.type);
+		}
+
+	};
+
+	this.validateProperties = function(properties) {
+		if (properties === undefined) {
+			this.errors.push(new UndefinedFieldIssue("properties"));
+		} else if (properties === null) {
+			this.errors.push(new NullFieldIssue("properties"));
+		} else if (properties === {}) {
+			this.errors.push(new EmptyObjectIssue("properties"));
+		} else {
+			this.validateTitle(properties.title);
+			this.validateLocation(properties.location);
+			var openingHours = properties.opening_hours;
+			var openingHoursUnclassified = properties.opening_hours_unclassified;
+			if (openingHours === null) {
+				this.validateOpeningHoursIsNull(openingHours);
+				this.validateOpeningHoursUnclassifiedIsNotNull(openingHoursUnclassified);
+			} else {
+				this.validateOpeningHours(openingHours);
+				this.validateOpeningHoursUnclassifiedIsNull(openingHoursUnclassified);
+			}
+		}
+	};
+
+	this.validateTitle = function(title) {
+		if (title === undefined) {
+			this.errors.push(new UndefinedFieldIssue("title"));
+		} else if (title === null) {
+			this.errors.push(new NullFieldIssue("title"));
+		} else if (title.length === 0) {
+			this.errors.push(new EmptyFieldIssue("title"));
+		}
+	};
+
+	this.validateLocation = function(location) {
+		if (location === undefined) {
+			this.errors.push(new UndefinedFieldIssue("location"));
+		} else if (location === null) {
```

Comment:
As discussed in #52, `location` shouldn't be mandatory, so a value of `null` should be allowed. It's fine to check for `undefined` and empty strings, though.